src/common/snippets/src/pass/hash.cpp

```diff
@@ -0,0 +1,487 @@
+            if (shape_str[0] == '[' && shape_str[shape_str.size() - 1] == ']')
+                shape_str = shape_str.substr(1, shape_str.size() - 2);
+            m_hash = hash_combine(hash_combine(m_hash, name), shape_str);
+        } else if (const auto& a = ov::as_type<ov::AttributeAdapter<ov::Dimension>>(&adapter)) {
+            const auto& attrs = a->get();
+            std::stringstream dim_str_stream;
+            dim_str_stream << attrs;
+            auto dim_str = dim_str_stream.str();
+            if (dim_str[0] == '{' && dim_str[dim_str.size() - 1] == '}')
+                dim_str = dim_str.substr(1, dim_str.size() - 2);
+            m_hash = hash_combine(hash_combine(m_hash, name), dim_str);
+        } else {
+            OPENVINO_THROW("Unsupported attribute type for serialization: ", name);
+        }
+    }
+
+    void on_adapter(const std::string& name, ov::ValueAccessor<bool>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::string>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<int64_t>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), static_cast<long long>(adapter.get()));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<double>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), adapter.get());
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_atribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<int64_t>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_atribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<uint64_t>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_atribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<float>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_atribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_hash = hash_combine(hash_combine(m_hash, name), create_atribute_list(adapter));
+    }
+    void on_adapter(const std::string& name, ov::ValueAccessor<std::shared_ptr<ov::Model>>& adapter) override {
+        if (name == "net") {
+            ovfunction_2_hash(m_hash, *adapter.get());
+        } else {
+            OPENVINO_THROW("Unsupported Model name.");
+        }
+    }
+};
+
+const std::unordered_map<ov::Node*, int> create_layer_ids(const ov::Model& model) {
```

Comment:
Minor: top-level const qualifier makes a little sense, since it doesn't prevent returned object from modification [details](https://stackoverflow.com/questions/1579435/should-useless-type-qualifiers-on-return-types-be-used-for-clarity). Also on L248

Reply:
done, thanks Ivan!